File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        Map<Integer, Integer> hashmap = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int n = target - nums[i];

            if (hashmap.containsKey(n)) {
                return new int[] {i, hashmap.get(n)};
            }
            else {
                hashmap.put(nums[i], i);
            }
        }
        return new int[2];
    }
}

File: 0001-two-sum/0001-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<Int32, Int32> map = new Dictionary<Int32, Int32>();

        for (int i = 0; i < nums.Length; i++) {
            int n = target - nums[i];

            if (map.ContainsKey(n)) {
                return new int[]{i, map[n]};
            }
            else {
                map.Add(nums[i], i);
            }
        }
        // shouldn't get here
        return new int[2];

    }
}